rs/src/parse/expressions/functions.rs

```diff
@@ -207,14 +195,45 @@ fn parse_function_argument(
     )
 }
 
+fn parse_function_option(
+    x: &substrait::FunctionOption,
+    y: &mut context::Context,
+) -> diagnostic::Result<FunctionOption> {
+    if x.preference.is_empty() {
```

Comment:
I think this will scream at you at runtime because `x.preference` is not traversed by a traversal macro. You have to do that or the resulting tree will not have all nodes in it. Or, more specifically, it *will* due to the fallback logic, but that logic will also throw a not-yet-implemented warning.